internal/provider/resource_vault_secrets_secret.go

```diff
@@ -0,0 +1,176 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+func NewVaultSecretsSecretResource() resource.Resource {
+	return &resourceVaultsecretsSecret{}
+}
+
+type resourceVaultsecretsSecret struct {
+	client *clients.Client
+}
+
+type Secret struct {
```

Comment:
Similar to above, something like `VaultSecretsSecret` might be better.

Reply:
done!